src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -276,112 +281,174 @@ private void CreateEmbeddedAttributesIfNeeded(DiagnosticBag diagnostics)
+                CreateAttributeIfNeeded(
                     ref _lazyIsByRefLikeAttribute,
                     diagnostics,
-                    AttributeDescription.IsByRefLikeAttribute);
+                    AttributeDescription.IsByRefLikeAttribute,
+                    CreateParameterlessEmbeddedAttributeSymbol);
             }
 
             if ((needsAttributes & EmbeddableAttributes.IsUnmanagedAttribute) != 0)
             {
-                CreateEmbeddedAttributeIfNeeded(
+                CreateAttributeIfNeeded(
                     ref _lazyIsUnmanagedAttribute,
                     diagnostics,
-                    AttributeDescription.IsUnmanagedAttribute);
+                    AttributeDescription.IsUnmanagedAttribute,
+                    CreateParameterlessEmbeddedAttributeSymbol);
             }
 
             if ((needsAttributes & EmbeddableAttributes.NullableAttribute) != 0)
             {
-                CreateEmbeddedNullableAttributeIfNeeded(
+                CreateAttributeIfNeeded(
                     ref _lazyNullableAttribute,
-                    diagnostics);
+                    diagnostics,
+                    AttributeDescription.NullableAttribute,
+                    CreateNullableAttributeSymbol);
             }
 
             if ((needsAttributes & EmbeddableAttributes.NullableContextAttribute) != 0)
             {
-                CreateEmbeddedNullableContextAttributeIfNeeded(
+                CreateAttributeIfNeeded(
                     ref _lazyNullableContextAttribute,
-                    diagnostics);
+                    diagnostics,
+                    AttributeDescription.NullableContextAttribute,
+                    CreateNullableContextAttributeSymbol);
             }
 
             if ((needsAttributes & EmbeddableAttributes.NullablePublicOnlyAttribute) != 0)
             {
-                CreateEmbeddedNullablePublicOnlyAttributeIfNeeded(
+                CreateAttributeIfNeeded(
                     ref _lazyNullablePublicOnlyAttribute,
-                    diagnostics);
+                    diagnostics,
+                    AttributeDescription.NullablePublicOnlyAttribute,
+                    CreateNullablePublicOnlyAttributeSymbol);
             }
         }
 
-        private void CreateEmbeddedAttributeIfNeeded(
-            ref SynthesizedEmbeddedAttributeSymbol symbol,
+        private SynthesizedEmbeddedAttributeSymbol CreateParameterlessEmbeddedAttributeSymbol(string name, NamespaceSymbol containingNamespace, DiagnosticBag diagnostics)
+            => new SynthesizedEmbeddedAttributeSymbol(
+                    name,
+                    containingNamespace,
+                    SourceModule,
+                    baseType: GetWellKnownType(WellKnownType.System_Attribute, diagnostics));
+
+        private SynthesizedEmbeddedNullableAttributeSymbol CreateNullableAttributeSymbol(string name, NamespaceSymbol containingNamespace, DiagnosticBag diagnostics)
```

Comment:
>CreateNullableAttributeSymbol [](start = 59, length = 29)

Consider inlining these `CreateNullable*AttributeSymbol` methods as lambdas, so the delegates are cached.